open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -190,6 +209,142 @@ def create_docker_host_vm(self, hackathon_id):
+                                  private_docker_api_port = args.private_docker_api_port,
+                                  container_count = 0,
+                                  container_max_count = args.container_max_count,
+                                  is_auto = AzureVMStartMethod.MANUAL,
+                                  state = args.state,
+                                  disabled = args.disabled,
+                                  create_time = self.util.get_now(),
+                                  update_time = self.util.get_now(),
+                                  hackathon_id = hackathon_id)
+        return ok()
+
+    def get_and_check_host_server(self, hackathon_id, host_server_id):
+        """
+        first get the docker host DB object for a hackathon.
+        and check whether the container_count is correct, if not, update this value in the database.
+
+        :param hackathon_id: the id of hackathon in DB
+        :type hackathon_id: Integer
+
+        :param host_server_id: the id of host_server in DB
+        :type host_server_id: Integer
+
+        :return: A object of docker_host_server
+        :rtype: DockerHostServer object or None
+        """
+        vm = self.db.find_first_object_by(DockerHostServer, id=host_server_id, hackathon_id=hackathon_id)
+        if vm is None:
+            self.log.warn('get docker_host fail, not find hostserver_id:' + host_server_id
+                          + 'hackathon_id:' + hackathon_id)
+            return None
+
+        vm_data = vm.dic()
+        try:
+            containers_url = 'http://%s:%d/containers/json' % (vm.public_dns, int(vm.public_docker_api_port))
+            res = requests.get(containers_url, timeout=5)
+            if res.status_code == 200:
+                jsonObject = json.loads(res.content)
+                if not vm_data["container_count"] == len(jsonObject):
+                    self.db.update_object(vm, container_count = len(jsonObject))
+                    vm_data["container_count"] = len(jsonObject)
+        except Exception as e:
+            self.log.error(e)
+
+        return vm_data
+
+    def update_host_server(self, hackathon_id, args):
+        """
+        update a docker host's information for a hackathon.
+
+        :param hackathon_id: the id of hackathon in DB
+        :type hackathon_id: Integer
+
+        :return: ok() if succeed.
+                 precondition_failed(...) if fail to update the docker_host's information
+        """
+        vm = self.db.find_first_object_by(DockerHostServer, id=args.id, hackathon_id=hackathon_id)
+        if vm is None:
+            self.log.warn('delete docker_host fail, not find hostserver_id:' + args.id
+                          + 'hackathon_id:' + hackathon_id)
+            return precondition_failed("", "host_server not found")
+
+        try:
+            ping_url = 'http://%s:%d/_ping' % (args.public_dns, int(args.public_docker_api_port))
```

Comment:
See this line in the same file multiple times. Duplicated codes is not a good coding style.  Whenever you find have to copy/paste your code with little changes, it means you have to optimize or refactor your code. Extract it to a method for example.